mail_tracking/models/mail_message.py

```diff
@@ -102,12 +142,58 @@ def _message_read_dict_postprocess(self, messages, message_tree):
         mail_message_ids = {m.get('id') for m in messages if m.get('id')}
         mail_messages = self.browse(mail_message_ids)
         partner_trackings = mail_messages.tracking_status()
-        email_cc = mail_messages._get_email_cc()
+        failed_message = mail_messages._get_failed_message()
         for message_dict in messages:
             mail_message_id = message_dict.get('id', False)
             if mail_message_id:
                 message_dict.update({
                     'partner_trackings': partner_trackings[mail_message_id],
-                    'email_cc': email_cc[mail_message_id],
+                    'failed_message': failed_message[mail_message_id],
                 })
+                message_dict['partner_trackings'] = \
+                    partner_trackings[mail_message_id]
         return res
+
+    @api.model
+    def _prepare_dict_failed_message(self, message):
```

Comment:
This would be better an `@api.multi` with `self.ensure_one()`. Passing an argument which is already a recordset of one `self` doesn't fit much with the purpose of `@api.model`.